### ISSUE TITLE
Empty union types don't pass through ak.argsort correctly

### ISSUE THREAD

**Issue by lgray** [scikit-hep/awkward#811]
Using the following file as input, because constructing a trivial example has proven painful:
https://drive.google.com/file/d/1a1Zla1htgjgF_b74Y0TKgcA3PABSpTFf/view?usp=sharing

You will need a recent version of coffea installed to run this (just pip install coffea).
I am using awkward==1.3.0rc1.

The following script:
```python3
from coffea import nanoevents
import awkward as ak

nanoevents.NanoAODSchema.mixins["FatJetLS"] = "PtEtaPhiMLorentzVector"

x = nanoevents.NanoEventsFactory.from_root(
    './nano106Xv8_on_mini106X_2017_mc_NANO_py_NANO_7.root',
    entry_start=100560, 
    entry_stop=102266
)

events = x.events()

muons = events.Muon
electrons = events.Electron
jets = events.Jet
fatjets = events.FatJet
subjets = events.SubJet
fatjetsLS = events.FatJetLS
met = events.MET

goodmuon = (
    (muons.mediumId)
    & (muons.miniPFRelIso_all <= 0.2)
    & (muons.pt >= 27)
    & (abs(muons.eta) <= 2.4)
    & (abs(muons.dz) < 0.1)
    & (abs(muons.dxy) < 0.05)
    & (muons.sip3d < 4)
)
good_muons = muons[goodmuon]

# electrons
goodelectron = (
    (electrons.mvaFall17V2noIso_WP90)
    & (electrons.pt >= 30)
    & (abs(electrons.eta) <= 1.479)
    & (abs(electrons.dz) < 0.1)
    & (abs(electrons.dxy) < 0.05)
    & (electrons.sip3d < 4)
)
good_electrons = electrons[goodelectron]

leptons = ak.concatenate([muons, electrons], axis=1)
good_leptons = ak.concatenate([good_muons, good_electrons], axis=1)

idx1 = ak.argsort(leptons.pt, axis=1)
idx2 = ak.argsort(good_leptons.pt, axis=1)

leptons[idx1]

good_leptons[idx2]
```

fails with
```
Traceback (most recent call last):
  File "test_unions.py", line 52, in <module>
    good_leptons[idx2]
  File "/Users/lagray/miniconda3/envs/coffea-jetmet/lib/python3.7/site-packages/awkward/highlevel.py", line 974, in __getitem__
    return ak._util.wrap(self.layout[where], self._behavior)
ValueError: only arrays of integers or booleans may be used as a slice
```

and indeed ak.argsort(good_leptons, axis=1) has some wacky type:
```python3
1706 * var * union[float32[parameters={"__doc__": "pt"}], float32[parameters={"__doc__": "p_{T}"}]]
```

The only difference between the path to idx1 and idx2 is that good_leptons is empty for this slice of events in this file.

**Comment by jpivarski** [comment]
@ianna, could you take a look at this? I don't know how `ak.argsort` ended up returning non-integers—I couldn't reproduce it with simpler examples than the one @lgray prepared.

**Comment by ianna** [comment]
@lgray - thanks for reporting it. The https://github.com/scikit-hep/awkward-1.0/pull/812 should fix it.

**Comment by ianna** [comment]
fixed via #812
